src/S3Bucket/S3BucketBrowser.vue

```diff
@@ -0,0 +1,111 @@
+<template>
+  <VSheet class="px-3 py-4" rounded>
+    <div class="d-inline-flex w-100">
+      <div class="w-25 d-inline-flex mt-1">
+        <VAvatar :image="avatarUrl" />
+        <span class="ml-3 text-h5">S3 Bucket Manager</span>
+      </div>
+      <VSelect 
+        v-if="!preLoadedResource"
+        label="Buckets:" 
+        :items="buckets"
+        item-title="name"
+        item-value="id"
+        return-object
+        class="w-75"
+        placeholder="Select S3 Bucket"
+        :hide-details="true"
+        @update:modelValue="getResourceSelection"
+      />
+    </div>
+    <VProgressLinear v-if="!bucketDetails && isLoading" indeterminate class="mt-3" color="blue-darken-4" rounded />
+    <BucketDisplay 
+      v-if="bucketDetails"
+      :api="api"
+      :is-flat="isFlat"
+      :location="bucketDetails?.location"
+      :resource="bucketDetails?.resource"
+      :state="bucketDetails?.state"
+      :update-resource-selection="updateResourceSelection"
+      @update:updateFlatten="() => isFlat = !isFlat" />
+  </VSheet>
+</template>
+
+<script setup>
+import { computed, onMounted, ref, watch } from "vue";
+import avatarUrl from '../assets/Icons_Storage_S3.png';
+import BucketDisplay from './BucketDisplay.vue';
+/**
+ * @typedef {Object} Props
+ * @property {ReturnType<import("@cloudbolt/js-sdk").createApi>} Props.api - The authenticated API instance
+ * @property {Object} Props.context - The applet context (on Resource Tab view contains current resource info)
+ */
+/** @type {Props} */
+const props = defineProps({
+  api: {
+    type: Object,
+    required: true,
+  },
+  context: {
+    type: Object,
+    required: true,
+  },
+});
+
+const buckets = ref()
+const bucketDetails = ref()
+const isFlat = ref()
+const isLoading = ref(false)
+const preLoadedResource = computed(() => buckets.value && props.context.resource && buckets.value.find((bucket) => bucket.name === props.context.resource.name))
+
+//  TODO CMP-54 Handle CSRF Token
+// Currently users must visit the dashboard before the CUI
+let token = sessionStorage.getItem("csrfToken");
+if (token) {
+  // eslint-disable-next-line vue/no-mutating-props
+  props.api.base.instance.defaults.headers.common['X-CSRFTOKEN'] = token
+} else {
+  console.log('Error, no token found. Please navigate to the dashboard to automatically set the token before returning to the CUI')
+}
+
+const getBuckets = async () => {
+  try {
+    const response = await props.api.base.instance.get('http://localhost:8001/ajax/s3-list-buckets/')
```

Comment:
I don't think you want the `http://localhost:8001` in here (or other calls either)

Reply:
OH RIGHT